### PR TITLE
Add om60 rev1 keyboard

### PR DESCRIPTION
**Pull request by jun10000**
## Description

Initial PR for adding om60 rev1 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
None

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/om60/config.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6A6E
+
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Jun
+#define PRODUCT         OM60
```
Comment (drashna) [inline keyboards/om60/config.h:30]:
```suggestion
```

File: keyboards/om60/config.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 8
+
+// wiring of each half
+#define MATRIX_ROW_PINS { D1, D0, D4, C6, D7 }
+#define MATRIX_COL_PINS { F4, F5, F6, F7, B1, B3, B2, E6 }
+#define UNUSED_PINS
```
Comment (drashna) [inline keyboards/om60/config.h:39]:
```suggestion
```

File: keyboards/om60/config.h [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      39,40,41,42,43, \
+      48,47,46,45,44, \
+      49,50,51,52,53, \
+      58,57,56,55,54, \
+      59,60,61,62,63, \
+      68,67,66,65,64  \
+  }
+#endif
+
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/om60/config.h:72]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/om60/config.h [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  #define USB_MAX_POWER_CONSUMPTION 100
+#endif
+
+#define TAP_CODE_DELAY 10
+
+#define ENCODERS_PAD_A { B4 }
+#define ENCODERS_PAD_B { B5 }
+#define ENCODERS 1
+#define ENCODERS_CCW_KEY { {6, 5} }
+#define ENCODERS_CW_KEY  { {6, 6} }
```
Comment (drashna) [inline keyboards/om60/config.h:104]:
```suggestion
```

File: keyboards/om60/encoder.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/* Copyright 2021 3araht
```
Comment (drashna) [inline keyboards/om60/encoder.c:1]: Please remove this file, and use encoder map instead. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

And via does support this.

File: keyboards/om60/info.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "keyboard_name": "OM60", 
+  "url": "https://github.com/jun10000", 
+  "maintainer": "jun10000",
```
Comment (drashna) [inline keyboards/om60/info.json:4]:
```suggestion
  "maintainer": "jun10000", 
  "usb": {
      "vid": "0x6A6E",
      "pid": "0x0001",
      "device_version": "0.0.1",
  }
```

File: keyboards/om60/keymaps/default/config.h [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define RGBLIGHT_SLEEP
```
Comment (drashna) [inline keyboards/om60/keymaps/default/config.h:21]: This should probably be moved to the keyboard's config.h

File: keyboards/om60/keymaps/via/config.h [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define RGBLIGHT_SLEEP
```
Comment (drashna) [inline keyboards/om60/keymaps/via/config.h:21]: Same

File: keyboards/om60/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```
Comment (drashna) [inline keyboards/om60/readme.md:27]:
```suggestion
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/om60/om60.h [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
+    E0L, E0R, \
```
Comment (drashna) [inline keyboards/om60/om60.h:24]:
```suggestion
```

File: keyboards/om60/om60.h [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    L04, L09, L14, L19, L24, L29, L34, L39, R44, R49, R54, R59, R64, R69  \
+  ) \
+  { \
+    {   L01,   L06,   L11,   L16,   L21,   L26,   L31,   L36 }, \
+    {   L02,   L07,   L12,   L17,   L22,   L27,   L32,   L37 }, \
+    {   L03,   L08,   L13,   L18,   L23,   L28,   L33,   L38 }, \
+    {   L04,   L09,   L14,   L19,   L24,   L29,   L34,   L39 }, \
+    {   L05,   L10,   L15,   L20,   L25,   L30,   L35,   L40 }, \
+    {   R41,   R46,   R51,   R56,   R61,   R66,   E0L, KC_NO }, \
+    {   R42,   R47,   R52,   R57,   R62,   R67,   E0R, KC_NO }, \
```
Comment (drashna) [inline keyboards/om60/om60.h:38]:
```suggestion
    {   R41,   R46,   R51,   R56,   R61,   R66,   KC_NO, KC_NO }, \
    {   R42,   R47,   R52,   R57,   R62,   R67,   KC_NO, KC_NO }, \
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/om60/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+*/
+
+#include QMK_KEYBOARD_H
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [2] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [3] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
```
Comment (drashna) [inline keyboards/om60/keymaps/via/keymap.c:27]: Heads up, you can fill these with transparent keycodes, just like the normal keymap. 

```suggestion
    [1] = { ENCODER_CCW_CW(_______, _______) },
    [2] = { ENCODER_CCW_CW(_______, _______) },
    [3] = { ENCODER_CCW_CW(_______, _______) },
```

**Review by drashna — APPROVED** [review]

File: keyboards/om60/info.json [hunk 13]
```diff
@@ -0,0 +1,15 @@
+{
+  "keyboard_name": "OM60",
```
Comment (fauxpark) [inline keyboards/om60/info.json:2]:
```suggestion
  "keyboard_name": "OM60", 
  "manufacturer": "jun10000",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by jun10000** [comment]
Thank you for fixing the encoder map feature.
It correctly works in the via (https://usevia.app/#/).

**Comment by jun10000** [comment]
@fauxpark
Is there any problems?

File: keyboards/om60/matrix.c [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+#elif (MATRIX_COLS <= 16)
+#    define print_matrix_header()  print("\nr/c 0123456789ABCDEF\n")
+#    define print_matrix_row(row)  print_bin_reverse16(matrix_get_row(row))
+#    define matrix_bitpop(i)       bitpop16(matrix[i])
+#    define ROW_SHIFTER ((uint16_t)1)
+#elif (MATRIX_COLS <= 32)
+#    define print_matrix_header()  print("\nr/c 0123456789ABCDEF0123456789ABCDEF\n")
+#    define print_matrix_row(row)  print_bin_reverse32(matrix_get_row(row))
+#    define matrix_bitpop(i)       bitpop32(matrix[i])
+#    define ROW_SHIFTER  ((uint32_t)1)
```
Comment (fauxpark) [inline keyboards/om60/matrix.c:34]:
```suggestion
#    define ROW_SHIFTER ((uint8_t)1)
#elif (MATRIX_COLS <= 16)
#    define ROW_SHIFTER ((uint16_t)1)
#elif (MATRIX_COLS <= 32)
#    define ROW_SHIFTER  ((uint32_t)1)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by jun10000** [comment]
@fauxpark
Are there another problems?

**Review by fauxpark — APPROVED** [review]

File: keyboards/om60/rules.mk [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
```
Comment (drashna) [inline keyboards/om60/rules.mk:21]: This should not be enabled at the keyboard level.   Please move this to the keymap level. 

```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**